core/src/main/scala/kafka/coordinator/GroupMetadata.scala

```diff
@@ -264,16 +266,21 @@ private[coordinator] class GroupMetadata(val groupId: String, initialState: Grou
   }
 
   def completePendingOffsetWrite(topicPartition: TopicPartition, offset: OffsetAndMetadata) {
-    offsets.put(topicPartition, offset)
+    if (!deletedPartitions.contains(topicPartition))
+      offsets.put(topicPartition, offset)
     pendingOffsetCommits.get(topicPartition) match {
```

Comment:
add a line break above, bit confusing as pendingOffsetCommits.get looks to be inside the if block